components/ILIAS/Contact/classes/class.ilMailSearchObjectGUI.php

```diff
@@ -369,31 +278,22 @@ public function cancel(): void
 
     public function showMembers(): void
     {
-        $obj_ids = [];
-        if ($this->http->wrapper()->query()->has('search_' . $this->getObjectType())) {
-            $obj_ids = $this->refinery->kindlyTo()->listOf(
-                $this->refinery->kindlyTo()->int()
-            )->transform(explode(',', (string) $this->http->wrapper()->query()->retrieve(
-                'search_' . $this->getObjectType(),
-                $this->refinery->kindlyTo()->string()
-            )));
-        } elseif ($this->http->wrapper()->post()->has('search_' . $this->getObjectType())) {
-            $obj_ids = $this->http->wrapper()->post()->retrieve(
-                'search_' . $this->getObjectType(),
-                $this->refinery->kindlyTo()->listOf(
-                    $this->refinery->kindlyTo()->int()
-                )
-            );
-        } elseif (ilSession::get('search_' . $this->getObjectType())) {
-            $obj_ids = $this->refinery->kindlyTo()->listOf(
-                $this->refinery->kindlyTo()->int()
-            )->transform(explode(',', (string) ilSession::get('search_' . $this->getObjectType())));
-            ilSession::set('search_' . $this->getObjectType(), '');
-        }
+        $this->tabs->clearTargets();
+        $this->tabs->setBackTarget(
+            $this->lng->txt('back'),
+            $this->ctrl->getLinkTarget($this, 'showMyObjects')
+        );
+
+        $obj_ids = $this->http->wrapper()->query()->has('contact_mailinglist_search_obj_ids')
```

Comment:
This is not a `dictOf`, but a `listOf`. Please also try to use `byTring` and `always` to avoid the `has` check.